### PR TITLE
KAFKA-16078: Be more consistent about getting the latest MetadataVersion

### PR DESCRIPTION
**Pull request by mumrah**
This PR creates MetadataVersion.latestTesting to represent the highest metadata version (which may be unstable) and MetadataVersion.latestProduction to represent the latest version that should be used in production. It fixes a few cases where the broker was advertising that it supported the testing versions even when unstable metadata versions had not been configured.

### CODE REVIEW

File: core/src/main/java/kafka/server/builders/LogManagerBuilder.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public class LogManagerBuilder {
     private int maxTransactionTimeoutMs = 15 * 60 * 1000;
     private ProducerStateManagerConfig producerStateManagerConfig = new ProducerStateManagerConfig(60000, false);
     private int producerIdExpirationCheckIntervalMs = 600000;
-    private MetadataVersion interBrokerProtocolVersion = MetadataVersion.latest();
+    private MetadataVersion interBrokerProtocolVersion = MetadataVersion.latest(true);
```
Comment (ijuma) [inline core/src/main/java/kafka/server/builders/LogManagerBuilder.java]: An enum instead of boolean would make this a lot more readable.

**Comment by cmccabe** [comment]
I agree with @ijuma that just having a boolean isn't very clear.

How about renaming `MetadataVersion.latest()` to `MetadataVersion.latestTesting()`, and then creating a separate function `MetadataVersion.latestProduction()`?

That seems like the clearest we're going to get with this. Devs won't want to invoke something with "testing" in the name in  a non-test context.

**Comment by cmccabe** [comment]
Alternately, if we wanted to make this literally impossible to invoke from a non-test context, we could move `MetadataVersion.latest` into `MetadataVersionTest.latest`.

The downside is that that would require all the other test modules to depend on `:server-common:test` (although a lot of them already do?) That might extend the build time? Not sure.

**Comment by mumrah** [comment]
All production usages now call `latestProduction` and test usages (including benchmark classes) call `latestTesting`. One exception is is `QuorumFeatures#defaultFeatureMap(boolean enableUnstable)` which calls either depending on its argument.

**Review by cmccabe — APPROVED** [review]
LGTM

**Pull request merged**